File: Codes/Level 2/Recursion and Backtracking/Permutations and Combinations/FactorPermutations.cpp
#include<iostream>
 #include<vector>
 using namespace std;
 
 // -----------------------------------------------------
 // This is a functional problem. Only this function has to be written
 // This function takes as input an integer
 // It should return the required output
 
 void factors(int n, vector<vector<int>> &ans, vector<int> currAns, int permn, int idx){
     if(n == 1){
         ans.push_back(currAns);
         return;
     }
     
     for(int i=idx; i<=n && i<permn; i++){
         if( n % i == 0 ){
             currAns.push_back(i);
             factors(n/i, ans, currAns, permn, i);
             currAns.pop_back();
         }
     }
 }
 
 vector<vector<int>> getFactors(int n) {
     //Write your code here
    vector<vector<int>> ans;
    vector<int> currAns;
     
    if(n == 1)  return ans;
    
    factors(n, ans, currAns, n, 2);
    
    return ans;
 }
 
 void display(vector<int>& res){
     cout<<"[";
     for(int i=0;i<res.size();i++){
         cout<<res[i];
         if(i!=res.size()-1){
             cout<<", ";
         }
     }
     cout<<"]"<<endl;
 }
 int main(int argc,char** argv){
     int n;
     cin>>n;
 
     vector<vector<int>> res=getFactors(n);
     for(vector<int> v:res){
         display(v);
     }
     
 }

File: Codes/Level 2/Recursion and Backtracking/Permutations and Combinations/FactorPermutations.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        List<List<Integer>> ans = getFactors(n);
        for(List<Integer> val : ans){
            System.out.println(val);
        }
    }

    public static List<List<Integer>> getFactors(int n){
        List<List<Integer>> ans = new ArrayList<>();
        if(n == 1)
            return ans;
        List<Integer> tempList = new ArrayList<>();

        factors(n, ans, tempList, n, 2);

        return ans;
    }

    private static void factors(int n, List<List<Integer>>  ans, List<Integer> tempList, int permn, int idx){
        if(n == 1){
            List<Integer> br = new ArrayList<>(tempList);
            ans.add(br);
            return;
        }
        for(int i=idx; i<=n && i < permn; i++){
            if(n % i == 0){
                tempList.add(i);
                factors(n/i, ans, tempList, permn, i);
                tempList.remove(tempList.size() - 1);
            }
        }
    }
}